content/300-guides/300-upgrade-guides/200-upgrading-versions/600-upgrading-to-prisma-5.mdx

```diff
@@ -0,0 +1,430 @@
+
+The use of `db push --force` can be replaced with the newer implementation `db push --accept-data-loss`.
+
+All other flags are from previous APIs and are no longer needed.
+
+### Removal of the <inlinecode>beforeExit</inlinecode> hook from the library engine
+
+Instead of the existing `beforeExit` hook we recommend using built-in Node.js exit events.
+
+The following code with Prisma 4:
+
+```js
+const exitHandler = () => {
+  // your exit handler code
+}
+
+prisma.$on('beforeExit', exitHandler)
+```
+
+Could become:
+
+```js
+const exitHandler = () => {
+  // your exit handler code
+}
+
+process.on('exit', exitHandler)
+process.on('beforeExit', exitHandler)
+process.on('SIGINT', exitHandler)
+process.on('SIGTERM', exitHandler)
+process.on('SIGUSR2', exitHandler)
+```
+
+### Removal of deprecated <inlinecode>prisma2</inlinecode> executable
+
+When Prisma 2 was released a new executable, `prisma2` was added to the CLI in addition to the existing `prisma` executable. Today we are removing the extra executable.
+
+If your scripts use `prisma2` with the Prisma CLI, please update `prisma2` to simply `prisma`.
+
+### Removal of deprecated <inlinecode>experimentalFeatures</inlinecode> property
+
+`previewFeatures` in a generator used to be called `experimentalFeatures`. We are finally removing that deprecated property.
+
+Update references of `experimentalFeatures` to `previewFeatures` manually or use the new code action in the Prisma VSCode extension.
+
+### Rename <inlinecode>migration-engine</inlinecode> to <inlinecode>schema-engine</inlinecode>
+
+The engine responsible for commands like `prisma migrate` and `prisma db` has been renamed from `migration-engine` to `schema-engine` to better describe its use. For many users, no changes will be required. However, if you need to explicitly include the engine file for any reason, you will need to update your code references.
+
+One example we have seen is projects using the Serverless Framework. In these instances, you may have rules such as:
+
+```yaml
+package:
+  patterns:
+    - '!node_modules/.prisma/client/libquery_engine-*'
+    - 'node_modules/.prisma/client/libquery_engine-rhel-*'
+    - '!node_modules/prisma/libquery_engine-*'
+    - '!node_modules/prisma/migration-engine-*'
+```
+
+The rule `'!node_modules/prisma/migration-engine-*'` rule will need to be updated to match the new naming.
+
+The [recommended rule from our documentation](https://www.prisma.io/docs/guides/deployment/deployment-guides/use-prisma-with-serverless-framework#lambda-functions-with-arm64-architectures) is not affected by this change.
+
+```yaml highlight=6;normal
+package:
+  patterns:
+    - '!node_modules/.prisma/client/libquery_engine-*'
+    - 'node_modules/.prisma/client/libquery_engine-rhel-*'
+    - '!node_modules/prisma/libquery_engine-*'
+    - '!node_modules/@prisma/engines/**'
+```
+
+## Upgrade the <inlinecode>prisma</inlinecode> and <inlinecode>@prisma/client</inlinecode> packages to Prisma 5
```

Comment:
Sorry if I missed the discussion on this, but shouldn't that be the first step, rather than a paragraph at the end? Most of what we are talking in the guide is not possible to do until you actually upgrade.

Reply:
I think that is just how it used to be in the old upgrade guides. No idea why. We can totally revisit that.

Reply:
Yup, this section and setup is based on previous upgrade guides. I'm happy to change this to be first, last, in the middle... whatever we decide is best 😊

Reply:
Can probably come before "Breaking Changes" indeed.

Reply:
Updated!